### ISSUE TITLE
[a11y]: Notification - inline - close button not tabbable

### ISSUE THREAD

**Issue by annawen1** [carbon-design-system/carbon#13616]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

latest

### Automated testing tool and ruleset

manual testing

### Description

1. The close button on the inline variation of `Notification` doesn't seem to be tabbable.
2. Ran into this while doing our v11 syncing work with Carbon Web Components - @oliviaflory pointed out the fill for the warning icons are not consistent and they have to have a black fill to pass color contrast.

<img width="822" alt="Screenshot 2023-04-17 at 5 20 21 PM" src="https://user-images.githubusercontent.com/54281166/232613846-b94b77a4-8229-4fff-8b6b-06e332320f7f.png">

### Reproduction/example

https://react.carbondesignsystem.com/?path=/story/components-notifications-inline--playground

### Steps to reproduce

Go to storybook and try tabbing into the inline notification to the close button

https://react.carbondesignsystem.com/?path=/story/components-notifications-inline--playground

For the icon fill issue, go to the storybook
1. Select inline notification variation
2. Set 'low-contrast' prop to true
3. select between the different warning "kind" variations (warning-alt and warning) and compare the icon fill colors

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by tay1orjones** [comment]
> inline variation of Notification doesn't seem to be tabbable

I know this feels goofy, but it's actually intentional. `InlineNotification` and `ToastNotification` only have limited role options:

https://github.com/carbon-design-system/carbon/blob/c6c83ce2be039c7ef95da70c34babd96c363e48d/packages/react/src/components/Notification/Notification.js#L390

Due to this they cannot contain interactive elements and do not receive focus. For this reason, the Close button is removed from the accessibility tree and is basically just an affordance for sighted users only.


`ActionableNotification` on the other hand gets `role=alertdialog`, allowing interactive elements within and the close button is tabbable there. It can also be configured as `inline`, mimicking the styling of `InlineNotification`

---

> the fill for the warning icons are not consistent and they have to have a black fill to pass color contrast

Wild, looks like this was the case back in v10 as well [link](https://v7-react.carbondesignsystem.com/?path=/story/components-notifications--toast&knob-ARIA%20role%20(role)=alert&knob-Action%20(NotificationActionButton%20%3E%20children)=Action&knob-Caption%20(caption)=00:00:00%20AM&knob-Duration%20in%20milliseconds%20to%20display%20notification%20(timeout)=0&knob-Icon%20description%20(iconDescription)=describes%20the%20close%20button&knob-Status%20icon%20description%20(statusIconDescription)=describes%20the%20status%20icon&knob-Subtitle%20(subtitle)=Subtitle%20text%20goes%20here.&knob-The%20notification%20kind%20(kind)=info&knob-Title%20(title)=Notification%20title&knob-Use%20low%20contrast%20variant%20(lowContrast)=true)

@oliviaflory @laurenmrice could you confirm if it's just the warning alt icon, or also the other icons as well - success, info, etc? Also this is isolated to just the `lowContrast` variants, correct?

**Comment by laurenmrice** [comment]
@tay1orjones All of the icons in the notifications for the high contrast and low contrast styles should have inner fills of either `black` or `white` for a11y color contrast reasons. I can audit those to see what needs to be updated in code.

**Comment by tay1orjones** [comment]
@laurenmrice Awesome! For lack of a better label, I put the `status: needs design spec` label on here.

**Comment by annawen1** [comment]
> > inline variation of Notification doesn't seem to be tabbable
> 
> I know this feels goofy, but it's actually intentional. `InlineNotification` and `ToastNotification` only have limited role options:
> 
> https://github.com/carbon-design-system/carbon/blob/c6c83ce2be039c7ef95da70c34babd96c363e48d/packages/react/src/components/Notification/Notification.js#L390
> 
> Due to this they cannot contain interactive elements and do not receive focus. For this reason, the Close button is removed from the accessibility tree and is basically just an affordance for sighted users only.
> 
> `ActionableNotification` on the other hand gets `role=alertdialog`, allowing interactive elements within and the close button is tabbable there. It can also be configured as `inline`, mimicking the styling of `InlineNotification`

Gotcha! Good to know the reasoning behind that! I'll adjust in our web components to match, thanks!

**Comment by laurenmrice** [comment]
These are what the inner fills should be for low and high contrast styles for the different themes.
<img width="936" alt="Frame 1" src="https://user-images.githubusercontent.com/43969356/232941609-30b933e4-72a3-4a7a-9fb4-321d327f128a.png">
